# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/WebIdentityHandler.h
# Commit: 46af55564501
# Full Hash: 46af555645013a126416a440b658f2969eca4b20
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 55c4afa3a46c -r 46af55564501 dom/credentialmanagement/identity/WebIdentityHandler.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/credentialmanagement/identity/WebIdentityHandler.h	Thu Jun 05 14:53:36 2025 +0000
@@ -0,0 +1,56 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_WebIdentityHandler_h
+#define mozilla_dom_WebIdentityHandler_h
+
+#include "mozilla/dom/AbortFollower.h"
+#include "mozilla/dom/CredentialManagementBinding.h"
+#include "mozilla/dom/Promise.h"
+#include "mozilla/dom/WebIdentityChild.h"
+#include "nsCycleCollectionParticipant.h"
+#include "nsPIDOMWindow.h"
+
+namespace mozilla::dom {
+
+class WebIdentityHandler final : public AbortFollower {
+ public:
+  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
+  NS_DECL_CYCLE_COLLECTION_CLASS(WebIdentityHandler)
+
+  explicit WebIdentityHandler(nsPIDOMWindowInner* aWindow) : mWindow(aWindow) {
+    MOZ_ASSERT(NS_IsMainThread());
+    MOZ_ASSERT(aWindow);
+  }
+
+  bool MaybeCreateActor();
+  void ActorDestroyed();
+
+  // AbortFollower
+  void RunAbortAlgorithm() override;
+
+  void GetCredential(const CredentialRequestOptions& aOptions,
+                     bool aSameOriginWithAncestors,
+                     const RefPtr<Promise>& aPromise);
+
+  void PreventSilentAccess(const RefPtr<Promise>& aPromise);
+
+  void Disconnect(const IdentityCredentialDisconnectOptions& aOptions,
+                  const RefPtr<Promise>& aPromise);
+
+  void SetLoginStatus(const LoginStatus& aStatus,
+                      const RefPtr<Promise>& aPromise);
+
+ private:
+  ~WebIdentityHandler();
+  nsCOMPtr<nsPIDOMWindowInner> mWindow;
+  RefPtr<WebIdentityChild> mActor;
+  RefPtr<Promise> mGetPromise;
+};
+
+}  // namespace mozilla::dom
+
+#endif  // mozilla_dom_WebIdentityHandler_h